src/config.rs

```diff
@@ -120,12 +135,21 @@ pub fn config_path() -> Result<PathBuf> {
 ///
 /// - Fails if the home directory cannot be found
 /// - Fails if the config directory cannot be created
-pub fn save_config(config: &Config) -> Result<()> {
-    let config_path = config_path()?;
-    let mut file = File::create(config_path)?;
-    let config = toml::to_string(config)?;
-    file.write_all(config.as_bytes())?;
-    Ok(())
+
+pub fn save_config(config: &Config) -> Result<(), ConfigError> {
+    // Tries to catch the config path
+    let config_path = config_path().map_err(|_| ConfigError::CreateConfigDir)?;
+    // Tries to create the config file
+    let mut file = File::create(&config_path).map_err(|_| ConfigError::CreateConfigFile)?;
+    // Tries to serialize the config to a string
+    let config = toml::to_string(config).map_err(|_| ConfigError::SaveConfigFile)?;
```

Comment:
This is not a "Save Config" error.
This is a conversion error to serialize the TOML config to string.

Should be a [`toml::ser::Error`](https://docs.rs/toml/latest/toml/ser/struct.Error.html)
which is what [`toml::to_string`](https://docs.rs/toml/latest/toml/fn.to_string.html) returns.

Can we also convert this with the `#[from]` and return the TOML error?

```suggestion
    let config = toml::to_string(config)?;
```